### PR TITLE
Polish RPM dependencies generator

### PR DESCRIPTION
**Pull request by rzhao271**
Ref https://github.com/microsoft/vscode/issues/17142

We used to have a hardcoded list of dependencies for the RPM package.
Then, I added a change that generates the rpm dependencies list per package CI run.
Now, I added some polish so that
- We have a reference list of dependencies, which means the CI can now detect when the dependencies list has changed.
- There is a constant that makes it relatively easy to switch between failing the build on new dependencies, and simply logging a warning.

### CODE REVIEW

File: build/linux/rpm/dep-lists.ts [hunk 1]
```diff
@@ -28,3 +28,92 @@ export const bundledDeps = [
 	'libvk_swiftshader.so',
 	'libffmpeg.so'
 ];
+
+export const referenceGeneratedDeps = [
```
Comment (deepak1556) [inline build/linux/rpm/dep-lists.ts:32]: List should be separate for each architecture.

**Comment by rzhao271** [comment]
[056d678ca2](https://github.com/microsoft/vscode/commit/056d678ca2d5edddb5d7388f3ca80e7dc68411b2) CI package build is green

**Comment by deepak1556** [comment]
Which CI run was used to extract the dependencies list into `build/linux/rpm/dep-lists.ts` ?

**Comment by rzhao271** [comment]
I manually copy-pasted from another run where I printed the dependencies:
https://dev.azure.com/monacotools/Monaco/_build/results?buildId=166357&view=results

**Review by deepak1556 — APPROVED** [review]

**Pull request merged**